plugins/org.jkiss.dbeaver.ui.navigator/src/org/jkiss/dbeaver/ui/navigator/actions/NavigatorHandlerObjectDelete.java

```diff
@@ -209,21 +211,20 @@ private static void changeDependentObjectsList(@NotNull DBRProgressMonitor monit
 
     @Override
     public void updateElement(UIElement element, Map parameters) {
-//        if (!updateUI) {
-//            return;
-//        }
-//        final ISelectionProvider selectionProvider = UIUtils.getSelectionProvider(element.getServiceLocator());
-//        if (selectionProvider != null) {
-//            ISelection selection = selectionProvider.getSelection();
-//
-//            if (selection instanceof IStructuredSelection && ((IStructuredSelection) selection).size() > 1) {
-//                element.setText(UINavigatorMessages.actions_navigator_delete_objects);
-//            } else {
-//                DBNNode node = NavigatorUtils.getSelectedNode(selection);
-//                if (node != null) {
-//                    element.setText(UINavigatorMessages.actions_navigator_delete_ + " " + node.getNodeTypeLabel()  + " '" + node.getNodeName() + "'");
-//                }
-//            }
-//        }
+        if (!updateUI) {
+            return;
+        }
+        final ISelectionProvider selectionProvider = UIUtils.getSelectionProvider(element.getServiceLocator());
+        if (selectionProvider != null) {
+            ISelection selection = selectionProvider.getSelection();
+
+            if (selection instanceof IStructuredSelection && ((IStructuredSelection) selection).size() == 1) {
+                DBNNode node = NavigatorUtils.getSelectedNode(selection);
+                if (node != null) {
+                    element.setText(String.format(
+                        "%s %s", UINavigatorMessages.actions_navigator_delete_, node.getNodeTypeLabel()));
```

Comment:
I kind of dislike the concatenation here. In some languages, the order is reversed (IIRC in Chinese, you say "Table Delete" instead of "Delete Table"). Let's replace it with an NLS pattern.